package.json

```diff
@@ -22,6 +22,7 @@
   },
   "dependencies": {
     "got": "^6.7.1",
+    "jest": "^24.9.0",
```

Comment:
`jest-cli` is already in `devDependencies`. This is not a runtime dependency.

Reply:
thanks eric, not sure how this got added but will remove it

Reply:
Odd, check failed "jest not found" although locally it passes the test. Any idea?

Reply:
That's strange. I saw nothing obvious by looking at the code. I will try with a clean copy of the project.

Reply:
I reproduced the error locally:

```
~/workspace/github/twilio/evanTheGreatWarrior/function-templates
$ npm test

> useful-twilio-functions@0.0.1 test /Users/eric/workspace/github/twilio/evanTheGreatWarrior/function-templates
> jest

sh: jest: command not found
npm ERR! Test failed.  See above for more details.
```

Reply:
thanks Eric

Reply:
I confirm that it works locally (unless the build forgets to run `npm install` somehow, like I did at first 😨).

Reply:
But it's there:

https://travis-ci.org/twilio-labs/function-templates/jobs/580138198#L221

Reply:
This part looks strange however:

```
$ npm install 

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules/fsevents):

npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

removed 64 packages in 5.069s
```

Reply:
Note: you committed changes to the `package-lock.json` in 0a65079a5b2.

Reply:
This looks related. Your `package-lock.json` must be in a modified state after removing `jest` from the dependencies in `package.json`. Can you commit it and push again?

Reply:
Thanks Eric let me give it a try :)

Reply:
Doesn't work locally now (same error, jest not found). So at least it's consistent lol.

Reply:
Anyway, i am playing locally with the jest module and indeed it works when I do npm install jest and it throws the error after npm uninstall jest. Not sure what I need to do to get it run without jest in the dependencies, any input welcome :)

Reply:
I also have the error after:

```
$ npm install jest
$ npm uninstall jest
$ npm install
$ npm test

> useful-twilio-functions@0.0.1 test /Users/eric/workspace/github/twilio/evanTheGreatWarrior/twilio/function-templates
> jest

sh: jest: command not found
npm ERR! Test failed.  See above for more details.
```

Reply:
The error disappears after I reset `node_modules`:

```
$ rm -rf node_modules
$ npm install
$ npm test

> useful-twilio-functions@0.0.1 test /Users/eric/workspace/github/twilio/evanTheGreatWarrior/twilio/function-templates
> jest

 PASS  video-token/tests/token.test.js
(...)
Ran all test suites.
```

Reply:
Good one Eric, seemed to work for me locally. Let me try and re-commit and let's see!

Reply:
Nope, I don't seem to get it right for any reason. Runs locally now without jest in the runtime dependencies. At this point, will leave it to the experts for further input when possible

Reply:
I compared the contents of the `node_modules` directory before and after the `npm install jest; npm uninstall jest` and found the following difference:

```
Only in function-templates-OK/node_modules/.bin: jest
```

This is a symbolic link which points on the `jest.js` script:

```
$ ls -l node_modules/.bin/jest
lrwxr-xr-x  1 eric  staff  23 Sep  3 15:26 node_modules/.bin/jest -> ../jest-cli/bin/jest.js
```

Running `npm install` again makes no change in `node_modules`, the symbolic link is still missing.

Reply:
I also tried `npm install --force`, to no avail.

Reply:
As you assumed, this is a cache issue. The `node_modules` folder is indeed cached by Travis:

```
[0Kattempting to download cache archive[0m
[32;1mfetching PR.45/cache-linux-xenial-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855--node-.tgz[0m
[32;1mfound cache[0m
travis_time:end:014fb458:start=1567516011891113778,finish=1567516013697557975,duration=1806444197
[0Ktravis_fold:end:cache.1
[0K
travis_fold:start:cache.npm
[0K
travis_time:start:2af6522e
[0Ktravis_time:end:2af6522e:start=1567516014320474532,finish=1567516014324477585,duration=4003053
[0Ktravis_time:start:1ead9463
[0Kadding /home/travis/build/twilio-labs/function-templates/node_modules to cache[0m
creating directory /home/travis/build/twilio-labs/function-templates/node_modules[0m
```

At some point, it got cached without the symbolic link, and now `npm install` never adds it back.

Reply:
cheers Eric. It's an odd case, but since it works for everyone else so far probably i am messing something :P I am thinking maybe to simply try a whole new branch + pull request since i have the function ready anyway, and see if it works

Reply:
Worth mentioning:

> There are only two hard things in Computer Science: cache invalidation and naming things.
> — Phil Karlton
> https://www.martinfowler.com/bliki/TwoHardThings.html